frontend/src/components/Bill/LocForm.jsx

```diff
@@ -19,20 +19,37 @@ const LocForm = ({ sponsorNames }) => {
     if (sessionPlaceName) {
       setPlaceName(sessionPlaceName);
     }
+
     const sessionRepresentatives =
       window.sessionStorage.getItem('representatives');
     if (sessionRepresentatives) {
-      setRepresentatives(JSON.parse(sessionRepresentatives));
+      const storedRepresentatives = JSON.parse(sessionRepresentatives);
+
+      // Filter state-level representatives
+      const stateRepresentatives = storedRepresentatives.filter(
+        (rep) => rep.jurisdiction === 'state'
+      );
+
+      // Update sponsorship status
+      const updatedSponsorshipStatus = {};
+      stateRepresentatives.forEach((rep) => {
+        updatedSponsorshipStatus[rep.name] = sponsorNames.includes(rep.name);
+      });
+
+      setRepresentatives(stateRepresentatives);
+      setSponsorshipStatus(updatedSponsorshipStatus);
     }
+
     const sessionLatitude = window.sessionStorage.getItem('latitude');
     if (sessionLatitude) {
       setLatitude(JSON.parse(sessionLatitude));
     }
+
     const sessionLongitude = window.sessionStorage.getItem('longitude');
     if (sessionLongitude) {
       setLongitude(JSON.parse(sessionLongitude));
     }
-  }, []);
+  }, [sponsorNames]);
 
   const reset = () => {
```

Comment:
Clicking "change location" throws an error: `storedRepresentatives.filter is not a function`, which is happening b/c the below is setting it to `{}` instead of `[]`.

I'd recommend simply deleting all of the items in sessionStorage instead of setting them to nullish values.